egs/wsj/s5/steps/nnet3/multilingual/combine_egs.sh

```diff
@@ -0,0 +1,140 @@
+cmd=run.pl
+minibatch_size=512      # it is the number of consecutive egs that we take from 
+                        # each source, and it only affects the locality of disk 
+                        # access. This does not have to be the actual minibatch size;
+num_jobs=10             # helps for better randomness across languages
+                        # per archive.
+samples_per_iter=400000 # this is the target number of egs in each archive of egs
+                        # (prior to merging egs).  We probably should have called
+                        # it egs_per_iter. This is just a guideline; it will pick
+                        # a number that divides the number of samples in the
+                        # entire data.
+lang2weight=            # array of weights one per input languge to scale example's output
+                        # w.r.t its input language during training.
+stage=0
+
+echo "$0 $@"  # Print the command line for logging
+
+if [ -f path.sh ]; then . ./path.sh; fi
+. parse_options.sh || exit 1;
+
+num_langs=$1
+
+shift 1
+args=("$@")
+megs_dir=${args[-1]} # multilingual directory
+mkdir -p $megs_dir
+mkdir -p $megs_dir/info
+if [ ${#args[@]} != $[$num_langs+1] ]; then
+  echo "$0: num of input example dirs provided is not compatible with num_langs $num_langs."
+  echo "Usage:$0 [opts] <num-input-langs,N> <lang1-egs-dir> ...<langN-egs-dir> <multilingual-egs-dir>"
+  echo "Usage:$0 [opts] 2 exp/lang1/egs exp/lang2/egs exp/multi/egs"
+  exit 1;
+fi
+
+required="egs.scp combine.scp train_diagnostic.scp valid_diagnostic.scp"
+train_scp_list=
+train_diagnostic_scp_list=
+valid_diagnostic_scp_list=
+combine_scp_list=
+
+# read paramter from $egs_dir[0]/info and cmvn_opts
+# to write in multilingual egs_dir.
+check_params="info/feat_dim info/ivector_dim info/left_context info/right_context info/frames_per_eg cmvn_opts"
+for param in $check_params; do
+  cat ${args[0]}/$param > $megs_dir/$param || exit 1;
+done
+cat ${args[0]}/cmvn_opts > $megs_dir/cmvn_opts || exit 1; # caution: the top-level nnet training
+                                                          # script should copy this to its
+                                                          # own dir.
+for lang in $(seq 0 $[$num_langs-1]);do
+  multi_egs_dir[$lang]=${args[$lang]}
+  for f in $required; do
+    if [ ! -f ${multi_egs_dir[$lang]}/$f ]; then
+      echo "$0: no such file ${multi_egs_dir[$lang]}/$f." && exit 1;
+    fi
+  done
+  train_scp_list="$train_scp_list ${args[$lang]}/egs.scp"
+  train_diagnostic_scp_list="$train_diagnostic_scp_list ${args[$lang]}/train_diagnostic.scp"
+  valid_diagnostic_scp_list="$valid_diagnostic_scp_list ${args[$lang]}/valid_diagnostic.scp"
+  combine_scp_list="$combine_scp_list ${args[$lang]}/combine.scp"
+
+  # check parameter dimension to be the same in all egs dirs
+  for f in $check_params; do
+    f1=$(cat $megs_dir/$f)
+    f2=$(cat ${multi_egs_dir[$lang]}/$f)
+    if [ "$f1" != "$f2" ]  ; then
```

Comment:
check the ivector_id as well
+ make sure the output directory inherits the ivector_id as well (I think now it does not, i.e. the nnet traiing script will complain about inconsistent use of ivector_ids).